### PR TITLE
examples module can be used as independent gradle project and extende…

### PR DESCRIPTION
**Pull request by paresh-p11**
…d further as is.
## Changes proposed in this pull request

* Moved SnappyTestRunner from snappy-core to snappy-examples
* Removed snappy-core dependency from examples/build.gradle file.
* Added snappy-examples dependency to core/build.gradle and dtests/build.gradle

## Patch testing

Did manual testing. Precheckin yet to be done

## ReleaseNotes.txt changes

(Does this change require an entry in ReleaseNotes.txt? If yes, has it been added to it?)

## Other PRs 

(Does this change require changes in other projects- store, spark, spark-jobserver, aqp? Add the links of PR of the other subprojects that are related to this change)

### CODE REVIEW

File: core/src/test/scala/io/snappydata/CommandLineToolsSuite.scala [hunk 1]
```diff
@@ -24,6 +24,7 @@ import org.apache.commons.io.output.TeeOutputStream
 import org.apache.spark.sql.collection.Utils
 
 import scala.sys.process._
+import io.snappydata.SnappyTestRunner
```
Comment (ashetkar) [inline core/src/test/scala/io/snappydata/CommandLineToolsSuite.scala]: Is this required? It's the same package.

File: examples/build.gradle [hunk 2]
```diff
@@ -24,7 +24,53 @@ sourceSets.main.java.srcDirs = []
...
+    sparkJobServerVersion = '0.6.2.9'
+  }
+
+  repositories {
+    mavenCentral()
+    maven { url 'https://dl.bintray.com/big-data/maven' }
+    maven { url "https://repo.spring.io/libs-release" }
+    maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
+    maven { url 'http://repository.snappydata.io/repository/internal' }
+    maven { url 'http://repository.snappydata.io/repository/snapshots' }
```
Comment (ashetkar) [inline examples/build.gradle]: Needs to be cleaned up. e.g. repository.snappydata.io does not exist any more.

File: examples/build.gradle [hunk 3]
```diff
@@ -24,7 +24,53 @@ sourceSets.main.java.srcDirs = []
...
+    maven { url 'https://app.camunda.com/nexus/content/repositories/public' }
+  }
+
+  dependencies {
+
+    compile 'org.scala-lang:scala-library:' + '2.11.8'
```
Comment (ashetkar) [inline examples/build.gradle]: Don't see these many dependencies in else block below. (log4j, slf4j, scala-library, scala-reflect, junit, snappy-core, spark-*). Doesn't snappy-cluster pull in those?

File: core/src/test/scala/io/snappydata/CommandLineToolsSuite.scala (hunk 1 above)
Comment (paresh-p11) [reply core/src/test/scala/io/snappydata/CommandLineToolsSuite.scala]: yes, that's redundant. Thanks

File: examples/build.gradle (hunk 2 above)
Comment (paresh-p11) [reply examples/build.gradle]: ok

**Comment by kneeraj** [comment]
The idea is that the user should be able to setup the gradle project in ide, idea and also eclipse and he should be able to run the examples, make changes and play with that.
Please complete a document task as to how the user can go about those in simple steps. Please explain with one of the bundled example.

**Pull request merged**